src/server/scripts/Commands/cs_modify.cpp

```diff
@@ -80,23 +80,32 @@ class modify_commandscript : public CommandScript
         return commandTable;
     }
 
-    //Edit Player HP
-    static bool HandleModifyHPCommand(ChatHandler* handler, const char* args)
+    template<typename... Args>
+    static void NotifyModification(ChatHandler* handler, Unit* target, const TrinityStrings resourceMessage, const TrinityStrings resourceReportMessage, Args&&... args)
+    {
+        if (Player* player = target->ToPlayer())
+        {
+            handler->PSendSysMessage(resourceMessage, handler->GetNameLink(player).c_str(), std::forward<Args>(args)...);
+            if (handler->needReportToTarget(player))
+                ChatHandler(player->GetSession()).PSendSysMessage(resourceReportMessage, handler->GetNameLink().c_str(), std::forward<Args>(args)...);
```

Comment:
Never forward a collapsing reference twice! For example forwarding a r-value qualified string two times will move the string the first time which means the string will be empty the second time you forward it.